enterprise/frontend/src/embedding-sdk/dev.md

```diff
@@ -0,0 +1,92 @@
+
+This is an _experimental_ command that should be much faster, it uses `tsc --incremental` to to generate the dts files and fixes them automatically by running the fixup script on watch.
+
+The `tsc` command will output a lot of errors, to keep the terminal output under control you may want to run the three different `embedding-sdk:dev:*` commands on different terminals.
+There is a VS code task named `Run embedding sdk dev commands` that does that
+
+## Storybook
+
+Storybook is a quick way of testing sdk components with hot reloading.
+
+Storybook expects an instance running on `localhost:3000` with some configuration:
+
+- on [/admin/settings/authentication/jwt](http://localhost:3000/admin/settings/authentication/jwt)
+  - Make sure "User Provisioning" is enabled.
+  - Set JWT secret to be "`0000000000000000000000000000000000000000000000000000000000000000`" in Admin > Authentication >
+    JWT > String used by the JWT signing key
+  - You will need to input a `JWT Identity Provider URI` even though it will not be used.
+  - Press "Save and enable"
+- on [/admin/settings/embedding-in-other-applications](http://localhost:3000/admin/settings/embedding-in-other-applications)
+  - Enable the sdk
+
+Then you can run `yarn storybook-embedding-sdk` to start storybook.
+
+Storybook will use the source files and not the built package.
+
+## E2E tests
+
+E2e tests for the sdk are currently done with cypress component tests.
+
+```bash
+yarn test-cypress-open-component-sdk
+```
+
+Cypress will use the built package, so we recommend running it together with either the watch or dev command.
+
+## Use the local build in a project
+
+To test a local build of the sdk in a local project, you can install the sdk from the resources folder.
+
+Assuming your projects are in a sibling directory the metabase core app, you can run
+
+```
+yarn add file:../metabase/resources/embedding-sdk
+```
+
+or
+
+```
+npm install --install-links  ../metabase/resources/embedding-sdk
+```
+
+### Common problems
+
+#### Caching issues
+
+Most bundlers optimize npm packages and cache the optimized files aggressively, here's how to clean the cache for the most used bundlers:
+
+next: `rm -rf .next`
+vite: `rm -rf node_modules/.vite`
+webpack: `rm -rf node_modules/.cache`
+
+We recommend clearing the cache every time you install the sdk, a common workflow is to:
+
+- remove `node_modules/@metabase` to make sure no old files are used
+- install the sdk with `yarn add file:../metabase/resources/embedding-sdk` / `npm install --install-links ../metabase/resources/embedding-sdk`
+- clean the cache of your bundler
+
+An example for vite: `rm -rf node_modules/@metabase && rm -rf node_modules/.vite && yarn add file:../metabase/resources/embedding-sdk`
+
+We usually create a custom `dev:link` script in the package.json to simplify the process.
+
+#### `Cannot read properties of null (reading 'useRef')`
+
+This happens when multiple versions of react are found in the project.
```

Comment:
This is my understanding of the problem at least, it messes up with symlinks and with npm projects nested in other projects

Reply:
Maybe we should include that symlink explanation so people might have some more context about it

Reply:
I included some more context, does it check out now?